### PR TITLE
[CKEditor4] refactor and expansion of ckeditor4 types

### PR DESCRIPTION
**Pull request by Enquier**
This refactor:
- Removes usage of Typescript classes which can cause unintended results
- Breaks up the definition file between the actual CKEditor global and the CKEDITOR type definitions to better align with usage patterns. 
- Provides for addition of plugins and user defined type extensions via several interfaces that can be extended.

Please fill in this template.

- [ X] Use a meaningful title for the pull request. Include the name of the package modified.
- [ X] Test the change in your own code. (Compile and run.)
- [ X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

### CODE REVIEW

**Comment by Enquier** [comment]
#1827

File: types/ckeditor4/index.d.ts [hunk 1]
```diff
@@ -1,2755 +1,58 @@
...
+/// <reference path="./plugins/cloudservices.d.ts" />
+/// <reference path="./plugins/dialog.d.ts" />
+/// <reference path="./plugins/elementspath.d.ts" />
+
+export default CKEDITOR;
+
+declare global {
+    interface Window {
+        CKEDITOR: CKEDITOR.CKEditorStatic;
+        CKEDITOR_BASEPATH: string;
```
Comment (stof) [inline types/ckeditor4/index.d.ts:56]: I think `CKEDITOR_BASEPATH` should be kept as `string | undefined`, as done before

**Review by sheetalkamat — COMMENTED** [review]
This is a huge change and needs review from owner to make sure it is appropriate.

File: types/ckeditor4/index.d.ts (hunk 1 above)
Comment (Enquier) [reply types/ckeditor4/index.d.ts:56]: Sounds good

**Comment by Enquier** [comment]
@sheetalkamat @gabbersepp who are the active owners that I should ping?

File: types/ckeditor4/index.d.ts (hunk 1 above)
Comment (Enquier) [reply types/ckeditor4/index.d.ts:56]: Updated today.

File: types/ckeditor4/index.d.ts [hunk 2]
```diff
@@ -1,2755 +1,58 @@
-// Type definitions for CKEditor 4.16
+// Type definitions for CKEDITOR 4.20
 // Project: https://ckeditor.com/ckeditor-4/, http://ckeditor.com
 // Definitions by: Thomas Wittwer <https://github.com/wittwert>
 //                 Stuart Long <https://github.com/stuartlong>
+//                 Viktor Pegy <https://github.com/viktorpegy>
```
Comment (sheetalkamat) [inline types/ckeditor4/index.d.ts:5]: Please remove this owner per https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63591#issuecomment-1466425870

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/ckeditor4/index.d.ts (hunk 2 above)
Comment (Enquier) [reply types/ckeditor4/index.d.ts]: Removed

**Review by gabritto — APPROVED** [review]

**Comment by Enquier** [comment]
Ready to merge

**Pull request merged**